src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -721,8 +721,9 @@ HRESULT CProfilerManager::LoadInstrumentationMethods(_In_ CConfigurationSource*
 }
 
 HRESULT CProfilerManager::RemoveInstrumentationMethod(
-    _In_ IInstrumentationMethod* pInstrumentationMethod)
+    _In_opt_ IInstrumentationMethod* pInstrumentationMethod)
```

Comment:
Why is this `_In_opt_` but then we fail the function if it's `nullptr`? Is this because the IDL file and generated headers state its `_In_opt_`?

Reply:
idl always generates _In_opt_. Potentially, it can't guarantee that the caller will not pass in a null pointer, so everything is marked as _In_opt_?

We are free to declare whatever contract we want, though, so it is fine to return an error.